docs/rfcs/003-open-in-atom.md

```diff
@@ -0,0 +1,137 @@
+
+![image](https://user-images.githubusercontent.com/378023/41271078-6c8e6a6c-6e49-11e8-96ee-5b0b5e55ef3d.png)
+
+### File
+
+1. Clones the repo (if needed)
+2. Checks out the branch
+3. Opens a new Atom window
+    - Or focuses the window if the repo already exists as an opened project
+4. Opens the file
+
+![image](https://user-images.githubusercontent.com/378023/41266930-30a24092-6e34-11e8-9c6e-36af97bc2bb3.png)
+
+### Line
+
+1. Clones the repo (if needed)
+2. Checks out the branch
+3. Opens a new Atom window
+    - Or focuses the window if the repo already exists as an opened project
+4. Opens the file
+5. Scrolls to the line
+6. Shows a blinking cursor at the beginning of the line 
+
+![open-in-atom](https://user-images.githubusercontent.com/378023/41338002-ccfe8944-6f2b-11e8-95d7-7e823e6d8643.png)
+
+
+## Drawbacks
+
+None that come to mind.
+
+## Rationale and alternatives
+
+By reducing the friction for moving from github.com to Atom, we can improve the following:
+
+- Higher quality of code reviews.
+    - Sometimes a reviewer might have the urge to suggest something, but they are not sure without testing it first. But since testing locally takes so many steps, they might just leave a comment like "Looks good to me" and move on. 👉 The easier it is to test changes, the better the reviews will become.
+- More contributions.
+    - If a reviewer tests something locally and makes edits, they might commit the changes and push it back to the PR. 👉 Reviewers that can easily edit and test a PR more likely become collaborators.
+- Better integration with other applications.
+    - Be able to open files, lines, merge conflicts etc. in Atom isn't just limited to github.com, it's also possible for other applications. For example if there is a merge conflict in GitHub Desktop, instead of manually switching and trying to find the conflict, GitHub Desktop could send you to Atom in a more seamless way.
+
+### Alternatives
+
+An alternative is to let users discover PRs/issues or get notified about changes **within** Atom. This would certainly be nice to have, but it will be a very long road until a "never have to leave Atom" is reached. Especially for our tiny team. We also don't have to pick one or the other and both approaches can complement each other.
+
+An alternative to have an Atom integration directly build into github.com would be to create a **browser extension**. A benefit is that it can be extended to other web sites/apps. For example an "Open in Atom" link inside Gmail for GitHub notifications. Downside: Users have to first install the extension.
+
+## Unresolved questions
+
+- [ ] Is github.com interested in adding an "Open in Atom" option? And possibly other editors? There is already a [Clone in Xcode](https://blog.github.com/2017-06-05-clone-in-xcode/) option.
+- [ ] Are there any security risks that need to be investigated beforehand.
+- [ ] What could the API (URI) look like? E.g. `atom://github/open?pull=1234`
```

Comment:
Hmm, let's see. How about:

```
atom://github/open/smashwilson%2Fremote-repo?pull=1234&line=456
```

And from an Enterprise instance, we'd encode the full remote URL:

```
atom://github/open/https%3A%2F%2Fgithub.enterprise.com%2Fsmashwilson%2Fremote-repo?ref=branchname&sha=abc123&file=some%2Fpath%2Fsource.js&line=456
```

Hard to say until we start implementing it though, I'm sure there's more info I won't realize we need until we're neck-deep in code 😆

Reply:
> we'd encode the full remote URL:

Or in all cases? Would this be possible: Keep it exactly the same as .com URLs, except replace `https://github.com` with 👉 `atom://github/open`

- .com URL: `https://github.com/atom/github/pull/1525/files#diff-7e96b28649369913c5e87f137813ade5R131`
- Atom: `atom://github/open/atom/github/pull/1525/files#diff-7e96b28649369913c5e87f137813ade5R131`

This would make it easier to turn a .com URL into and Atom URI. By hand or by other apps, browser extensions etc.

Also, as a first step, Atom could offer an "Open from GitHub URL" like in [Visual Studio](https://github.com/github/VisualStudio/pull/1763). Then later consider adding "Open in Atom" buttons to .com for an even better UX.

Reply:
Or even simpler:
```
atom://github?open=https://github.com/atom/github/pull/1525/files#diff-7e96b28649369913c5e87f137813ade5R13
```
You'd need to escape the `#`, but apart from that I think it would work.